### ISSUE TITLE
UBL issues

### ISSUE THREAD

**Issue by rlcsoft** [mriscoc/Ender3V2S1#574]
UBL is miss calculating bed size 

Center is correct when homing 
Y axis is correct
X axis is wrong when bed tramming and leveling.

it’s only doing the left side of the bed.

the calculation is incorrect for UBL 

defaulted the settings, rolled firmware from dec to October still has that issue 

![5B0E93FD-D8DB-4332-85BA-0DD7E163073A](https://user-images.githubusercontent.com/40546304/209421846-54d0ed8a-1de7-4277-aa57-da62ff310fc2.jpeg)
![E2A0D7B5-A68D-4C97-85B8-436A5B1591D4](https://user-images.githubusercontent.com/40546304/209421847-565a04f0-c373-47a4-9492-f26f43ffdc08.jpeg)

Firmware version and release date
- last two versions I have tested 
**Additional context**

**Comment by mriscoc** [comment]
Did you maximize and center the mesh insets? https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides

**Comment by rlcsoft** [comment]
I Followed the instructions, leveled the bed. Reset the firmware and ran G502 and it moved the grid to the center but ubl isn't working on 9x9 with HS off 
that second of my bed 

<img width="697" alt="Screenshot 2022-12-24 201320" src="https://user-images.githubusercontent.com/40546304/209454443-b9b6aaa8-aba1-4bbd-a3e1-5cf095d9f6e4.png">
<img width="697" alt="Screenshot 2022-12-24 201320" src="https://user-images.githubusercontent.com/40546304/209454506-30781ab3-db7a-4654-9a13-f9d3a7d992ac.png">

**Comment by rlcsoft** [comment]
![IMG_3875](https://user-images.githubusercontent.com/40546304/209454543-c60b51da-3efa-4916-ad1b-d295a1825551.jpg)
![IMG_3885](https://user-images.githubusercontent.com/40546304/209454544-ee8519c7-3c09-4d24-8fd4-d67d87677d1c.jpg)
![IMG_3886](https://user-images.githubusercontent.com/40546304/209454545-58bc5e62-8e96-40f5-83a3-81addf7a7a04.jpg)

**Comment by rlcsoft** [comment]
![321758672_916941609469901_66008525258556251_n](https://user-images.githubusercontent.com/40546304/209454558-b8df0ee6-3c49-4123-b21b-cd547ef5d9dc.jpg)
![IMG_3857](https://user-images.githubusercontent.com/40546304/209454561-43115819-ed5b-4ee7-9679-c5640214ebe1.jpg)

**Comment by rlcsoft** [comment]
Can you reach out to me via email please : rlcsoft@gmail.com,

**Comment by mriscoc** [comment]
I just saw an issue introduced by recent changes in the G30 code of Marlin that could have an effect in the determination of the probe positions. I will need some time to investigate if it is related.

**Comment by mriscoc** [comment]
I did some probe positioning tests and were successful, mesh is correctly centered after the maximize/center procedure.
Re-check you probe offsets values, maximize and center must be the last step.

**Comment by rlcsoft** [comment]
I swapped CR Touch for BL Touch, testing now with UBL special config compile. I noticed different readings.

Will keep you updated.

**Comment by rlcsoft** [comment]
Results:
The Swapped to a BL-Touch fixed the readings on the bed(tried another new CR-Touch) 
9x9 mesh area was small, 7x7 was more spread out went further on the bed. 

Need advise on spreading the probe zone to 15mm from edges.  How can 9x9 be fixed. 

![image1](https://user-images.githubusercontent.com/40546304/209499942-deeaa426-07f4-4798-aeec-772b5c3e71a1.jpeg)
![image0](https://user-images.githubusercontent.com/40546304/209499943-03afcf59-f006-4fe6-84dd-daec59bf074f.jpeg)

**Comment by mriscoc** [comment]
Some probing positions are unreachable and that depends on the probe offset. With lower probe offsets you must be able to decrease the margins and increase mesh area. If you don't need a centered mesh area just select maximize area, the printer will use the maximum reachable probing area available.

**Comment by rlcsoft** [comment]
Cr touch was confirmed issue.

**Comment by daninden** [comment]
> Cr touch was confirmed issue.

is it easy to swap the cr touch for a bl touch?

**Comment by rlcsoft** [comment]
Bl touch…. Work for a while then started again leveling issues. It makes no sense Can you email me so I can go over what I ***@***.*** Sent from my iPhoneOn Jan 3, 2023, at 7:18 AM, Dan Inden ***@***.***> wrote:﻿

Cr touch was confirmed issue.

is it easy to swap the cr touch for a bl touch?

**Comment by Tempestine** [comment]
I've been having issues with UBL on my Ender 3 V2 with a 4.2.7 board since updating to the lastest professional firmware (https://github.com/mriscoc/Ender3V2S1/releases/download/20221222/Ender3V2-427-BLTUBL-MPC-20221222.bin) firmware. The mesh would map strange and when I printed a bed level test, one side was squished and the other side would not stick. I thought it was perhaps not applying the mesh. I think I solved it by compiling my own firmware from the source. The mesh looks totally different now. I think there may be something wrong with some of the precompiled firmwares.
